app.rb

```diff
@@ -0,0 +1,77 @@
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'book'
+require_relative 'rental'
+
+class App
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def list_all_books
+    puts 'List of all books:'
+    @books.each { |book| puts "#{book.title} by #{book.author}" }
+  end
+
+  def list_all_people
+    puts 'List of all people:'
+    @people.each { |person| puts person.name }
+  end
+
+  def create_person(age, name, role, options = {})
+    case role
+    when 'teacher'
+      specialization = options[:specialization]
+      person = Teacher.new(@people.length + 1, age, name, specialization)
+    when 'student'
+      classroom = options[:classroom]
+      person = Student.new(@people.length + 1, age, name, classroom)
+    else
+      puts 'Invalid role. Person not created.'
+      return
+    end
+
+    @people << person
+    puts "#{person.name} has been created as a #{role}."
+  end
+
+  def create_book(title, author)
+    book = Book.new(title, author)
+    @books << book
+    puts "#{book.title} by #{book.author} has been created."
+  end
+
+  def create_rental(date, person_id, book_title)
+    person = find_person_by_id(person_id)
+    book = find_book_by_title(book_title)
+
+    return unless person && book
+
+    rental = Rental.new(date, book, person)
+    @rentals << rental
+    puts "#{person.name} has rented '#{book.title}' by #{book.author} on #{date}."
+  end
```

Comment:
- [ ] After creating people or listing people it is not possible to check the ID, so it is not possible to create the rental correctly. Please 🙏🏽 let the user know the ID of people to create rentals successfully. Additionally, make sure that your app matches the original design please 🙏🏽 check the [example video](https://www.youtube.com/watch?v=vkkgrhD6aXQ) and add the list options to select people and books by number of the list like in the original design.

Reply:
Hi @Innocent259, there is still an error when I try to create a Rental, did you experience this error from your end? 
![Screenshot_166](https://github.com/Innocent259/School-library/assets/69638013/b5a19285-22ca-4e29-b62a-e5ce4a544f16)